### PR TITLE
make Mediaviewer work in OCIS+Reva

### PR DESCRIPTION
**Pull request by madsi1m**
## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: apps/media-viewer/src/Mediaviewer.vue [hunk 1]
```diff
@@ -207,20 +207,19 @@ export default {
         a: 1
       })
 
+      this.activeMediaFile.path = this.activeMediaFile.path.replace(/^\//, '');
       if (this.publicPage()) {
         const path = [
           '..',
-          'dav',
-          'public-files',
```
Comment (DeepDiver1975) [inline apps/media-viewer/src/Mediaviewer.vue]: This will break on public shares

File: apps/media-viewer/src/Mediaviewer.vue [hunk 2]
```diff
@@ -207,20 +207,19 @@ export default {
...
+          'webdav',
           this.activeMediaFile.path
         ].join('/')
 
         return this.$client.files.getFileUrl(path) + '?' + query
       }
       const path = [
         '..',
-        'dav',
```
Comment (DeepDiver1975) [inline apps/media-viewer/src/Mediaviewer.vue]: Is the dav endpoint not available on reva/ocis?

File: apps/media-viewer/src/Mediaviewer.vue (hunk 1 above)
Comment (madsi1m) [reply apps/media-viewer/src/Mediaviewer.vue]: a closer look implies that for some reason `this.publicPage()` is true for non public files all the time in reva/ocis

File: apps/media-viewer/src/Mediaviewer.vue (hunk 2 above)
Comment (madsi1m) [reply apps/media-viewer/src/Mediaviewer.vue]: not yet (soon?)

File: apps/media-viewer/src/Mediaviewer.vue (hunk 1 above)
Comment (madsi1m) [reply apps/media-viewer/src/Mediaviewer.vue]: Is something missing in ocs @butonic ?

**Comment by PVince81** [comment]
I seem to remember that the media viewer is now working, now that the ocis-thumbnailer was developed/fixed. @C0rby can you confirm ?

in any case we need to fix the backend (if not already) to also generate thumbnails on those dav endpoints

**Comment by C0rby** [comment]
> I seem to remember that the media viewer is now working, now that the ocis-thumbnailer was developed/fixed. @C0rby can you confirm ?

Yes, but we need to update the `ocis-thumbnails` dependency in `ocis` because there was a bug which prevented the generation of thumbnails.

Also because the thumbnail service now uses the predefined sizes, the mediaviewer only gets images up until the max predefined size. So we may have to add some more like "1920x1080" for FHD images.

**Comment by C0rby** [comment]
The mediaviewer is working for me locally.
Is this pr still relevant?

**Comment by butonic** [comment]
fixed using pre signed urls

**Pull request closed without merging**